src/shared/s3_manager.py

```diff
@@ -0,0 +1,235 @@
+import csv
+import logging
+import os
+import traceback
+
+import awswrangler
+import boto3
+import numpy
+import pandas
+
+from shared import (
+    awswrangler_functions,
+    enums,
+    functions,
+)
+
+log_level = os.environ.get("LAMBDA_LOG_LEVEL", "INFO")
+logger = logging.getLogger()
+logger.setLevel(log_level)
+
+
+class S3Manager:
+    """Class for managing S3 paramaters/access from data in an AWS SNS event.
+
+    This is generally intended as a one stop shop for the data processing phase
+    of the aggregator pipeline, providing commmon file paths/sns event parsing helpers/
+    stripped down write methods. Consider adding utility functions here instead of using
+    raw awswrangler/shared functions to try and make those processes simpler.
+    """
+
+    def __init__(self, event):
+        self.s3_bucket_name = os.environ.get("BUCKET_NAME")
+        self.s3_client = boto3.client("s3")
+        self.sns_client = boto3.client("sns", region_name=self.s3_client.meta.region_name)
+        self.event_source = event["Records"][0]["Sns"]["TopicArn"]
+        self.s3_key = event["Records"][0]["Sns"]["Message"]
+        s3_key_array = self.s3_key.split("/")
+        self.study = s3_key_array[1]
+        self.data_package = s3_key_array[2].split("__")[1]
+        self.site = s3_key_array[3]
+        self.version = s3_key_array[4].split("__")[-1]
+        self.metadata = functions.read_metadata(
+            self.s3_client, self.s3_bucket_name, meta_type=enums.JsonFilename.TRANSACTIONS.value
+        )
+        self.types_metadata = functions.read_metadata(
+            self.s3_client,
+            self.s3_bucket_name,
+            meta_type=enums.JsonFilename.COLUMN_TYPES.value,
+        )
+        self.parquet_aggregate_path = (
+            f"s3://{self.s3_bucket_name}/{enums.BucketPath.AGGREGATE.value}/"
+            f"{self.study}/{self.study}__{self.data_package}/"
+            f"{self.study}__{self.data_package}__{self.version}/"
+            f"{self.study}__{self.data_package}__aggregate.parquet"
+        )
+        self.csv_aggregate_path = (
+            f"s3://{self.s3_bucket_name}/{enums.BucketPath.CSVAGGREGATE.value}/"
+            f"{self.study}/{self.study}__{self.data_package}/"
+            f"{self.version}/"
+            f"{self.study}__{self.data_package}__aggregate.csv"
+        )
+        # TODO: Taking out a folder layer to match the depth of non-site aggregates
+        # Revisit when targeted crawling is implemented
```

Comment:
To expand on this a bit here: [this bit](https://github.com/smart-on-fhir/cumulus-aggregator/blob/7e72e89205a0e25afd304cff7f90139e78bad223/template.yaml#L771) of the glue config specifies that table names will be taken from the fifth directory down. Since we're introducing site as a concept here, adding a new layer, I elected to strip one element out of the "ideal" path. 

#125 discusses a way around this, which basically amounts to creating a crawler on demand and pointing it at just the directory you want to crawl new data in.